ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+            color: partial_vec_to_u8_with_reverse(&raw_glass_details[4..6]),
+            scale: partial_vec_to_u8_with_reverse(&raw_glass_details[7..10]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_glass_details[11..15]),
+        };
+
+        let mole_details = MoleDetails {
+            is_enabled: raw_mole_details[0],
+            scale: partial_vec_to_u8_with_reverse(&raw_mole_details[1..4]),
+            x_position: partial_vec_to_u8_with_reverse(&raw_mole_details[5..9]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_mole_details[10..14]),
+        };
+
+        MiiData {
+            mii_options: options,
+            mii_selector_position: position,
+            mii_console_identity: device,
+            system_id,
+            mac_address: creator_mac,
+            mii_details: details,
+            mii_name,
+            height,
+            width,
+            face_style,
+            face_details,
+            hair_style,
+            hair_details,
+            eye_details,
+            eyebrow_details,
+            nose_details,
+            mouth_details,
+            mustache_details,
+            beard_details,
+            glass_details,
+            mole_details,
+            author_name,
+        }
+    }
+}
+
+// Methods to handle "_bits_", ``bitvec`` cannot compile to 32-bit targets, so I had to create a few
+// helper methods
+
+/// Transforms a u8 into a [bool; 8]
+fn vec_bit(data: u8) -> [bool; 8] {
+    (0..8)
+        .map(|i| (data & (1 << i)) != 0)
+        .collect::<Vec<bool>>()
+        .try_into()
+        .unwrap()
+}
+
+/// Transforms a [bool; 8] into an u8
+fn vec_bit_to_u8(data: [bool; 8]) -> u8 {
+    let mut result: u8 = 0;
+    data.map(u8::from).iter().for_each(|&bit| {
+        result <<= 1;
+        result ^= bit;
+    });
+    result
+}
+
+/// The reverse allows to write things on a more _humane_ way, but return a LE u8
+fn partial_vec_to_u8_with_reverse(data: &[bool]) -> u8 {
+    let leading_zeroes_to_add = 8 - data.len();
+    let leading_zeroes = vec![false; leading_zeroes_to_add];
+    let mut val = [data, leading_zeroes.as_slice()].concat();
+    val.reverse();
+    vec_bit_to_u8(val.try_into().unwrap())
+}
+
+/// UTF-16 Strings are give in pairs of bytes (u8), this converts them into an _actual_ string
+fn utf16_byte_pairs_to_string(data: &[u8]) -> String {
+    let raw_utf16_composed = data
+        .chunks(2)
+        .collect::<Vec<&[u8]>>()
+        .iter()
+        .map(|v| u16::from_le_bytes([*v.first().unwrap_or(&0), *v.get(1).unwrap_or(&0)]))
+        .collect::<Vec<u16>>();
+
+    String::from_utf16(raw_utf16_composed.as_slice())
+        .unwrap()
+        .replace('\0', "")
+}
+
+/// Gets the values from the slice and concatenates them
+fn get_and_concat_vec_bit(data: &[u8], get_values: &[usize]) -> Vec<bool> {
+    get_values
+        .iter()
+        .map(|v| vec_bit(data[*v]))
+        .collect::<Vec<[bool; 8]>>()
+        .concat()
+}
```

Comment:
nit: This could be simpler:
```suggestion
fn get_and_concat_vec_bit(data: &[u8], get_values: &[usize]) -> Vec<bool> {
    get_values
        .iter()
        .flat_map(|v| vec_bit(data[*v]))
        .collect()
}
```